### PR TITLE
Ensure that dollar signs are properly escaped in substitutions.

### LINKED ISSUE #36: $ in replacement strings gets interpreted as a part of the regex replacement
$ should be escaped to prevent unintended sequences like $1 or $& getting substituted from parts of the template.

### PR DESCRIPTION
**Pull request by ljharb**
Fixes #36.

### CODE REVIEW

File: lib/polyglot.js [hunk 1]
```diff
@@ -301,10 +301,13 @@
   function interpolate(phrase, options) {
     for (var arg in options) {
       if (arg !== '_' && options.hasOwnProperty(arg)) {
+        // ensure replacement value is escaped to prevent special $-prefixed regex replace tokens
+        // the "$$$$" is needed because each "$" needs to be escaped with "$" itself, and we need two in the resulting output.
+        var replacement = typeof options[arg] === 'string' ? String.prototype.replace.call(options[arg], /\$/g, '$$$$') : options[arg];
```
Comment (spikebrehm) [inline lib/polyglot.js]: Cache the regexp in a variable at the top of the file?

File: lib/polyglot.js [hunk 2]
```diff
@@ -301,10 +301,13 @@
   function interpolate(phrase, options) {
     for (var arg in options) {
       if (arg !== '_' && options.hasOwnProperty(arg)) {
+        // ensure replacement value is escaped to prevent special $-prefixed regex replace tokens
```
Comment (spikebrehm) [inline lib/polyglot.js]: s/ensure/Ensure/

File: lib/polyglot.js (hunk 1 above)
Comment (spikebrehm) [inline lib/polyglot.js]: Long line :)

**Comment by spikebrehm** [comment]
Nice test.

File: lib/polyglot.js (hunk 1 above)
Comment (ljharb) [inline lib/polyglot.js]: good call, i should cache `String#replace` also. Will update in a bit.

File: lib/polyglot.js (hunk 2 above)
Comment (ljharb) [inline lib/polyglot.js]: complete, balanced nutrition

**Comment by ljharb** [comment]
Updated!

File: lib/polyglot.js [hunk 3]
```diff
@@ -298,13 +300,21 @@
   //
   // Does the dirty work. Creates a `RegExp` object for each
   // interpolation placeholder.
+  var dollarRegex = /\$/g;
+  var dollarBillsYall = '$$$$';
```
Comment (spikebrehm) [inline lib/polyglot.js]: lol

File: lib/polyglot.js [hunk 4]
```diff
@@ -298,13 +300,21 @@
...
   function interpolate(phrase, options) {
     for (var arg in options) {
       if (arg !== '_' && options.hasOwnProperty(arg)) {
+        // Ensure replacement value is escaped to prevent special $-prefixed regex replace tokens
```
Comment (spikebrehm) [inline lib/polyglot.js]: NBD but it would be nice to truncate comments at ~80 chars

File: lib/polyglot.js [hunk 5]
```diff
@@ -298,13 +300,21 @@
...
+        // the "$$$$" is needed because each "$" needs to be escaped with "$" itself, and we need two in the resulting output.
+        var replacement = options[arg];
+        if (typeof options[arg] === 'string') {
```
Comment (spikebrehm) [inline lib/polyglot.js]: s/options[arg]/replacement/

File: lib/polyglot.js (hunk 4 above)
Comment (ljharb) [inline lib/polyglot.js]: fixed

File: lib/polyglot.js (hunk 5 above)
Comment (ljharb) [inline lib/polyglot.js]: fixed

**Pull request merged**